src/tools/create-file-stasher.js

```diff
@@ -140,9 +149,13 @@ const createFileStasher = input => {
                 fileContentType
               );
             };
+
             if (isStreamed) {
               maybeResponse.throwForStatus();
-              return maybeResponse.buffer().then(parseFinalResponse);
+              return maybeResponse.buffer().then(buffer => {
+                maybeResponse.dataBuffer = buffer;
+                return parseFinalResponse(maybeResponse);
```

Comment:
As I'm looking at this diff, it seems that before we were passing `parseFinalResponse` a buffer object, but not the complete response object. With the change, now we will be giving a complete response object. Am I understanding that correctly? If so, does that have a side effect for other scenarios where we were streaming files?

Mainly concerned with the fact that we'll now be reading off the headers of the response, where as before we didn't touch them. Seems like before and after, we are still leaving the buffer with the response body untouched, so thinking this will be ok.

Reply:
Yes, you understand it correctly. Without this change, we were always ignoring the response headers if the stream mode is on (`z.request({ raw: true })`). I'm not sure about the side effect, though. Do you have an example?

Reply:
Not specifically. I was thinking it might mess up `.pip()` or `.read()` if somebody was doing that off the response object, but, we've only been passing in the body this whole time, so it shouldn't matter.